lib/extract_field/FieldExtractor.js

```diff
@@ -0,0 +1,66 @@
+const querystring = require('querystring');
+const { sha256, accessNestedObjectValueByStringPath } = require('../pxutil');
+
+const SENT_THROUGH = { BODY: 'body', QUERY_PARAM: 'query-param', HEADER: 'header' };
+const ENCODING_TYPE = { URL_ENCODE: 'url-encode', CLEAR_TEXT: 'clear-text', BASE64: 'base64', CUSTOM: 'custom' };
+
+class FieldExtractor {
+    constructor(sentThrough, contentType, encoding, fieldsToExtract) {
+        this.containerName = this._getContainerName(sentThrough);
+        this.contentType = contentType;
+        this._decode = this._getDecodeFunction(encoding);
+        this.fieldsToExtract = fieldsToExtract instanceof Array ? fieldsToExtract : [];
```

Comment:
Dont you think it should be:

this.fieldsToExtract = fieldsToExtract instanceof Array ? fieldsToExtract : [fieldsToExtract];

??